Marlin/src/HAL/AVR/MarlinSerial.cpp

```diff
@@ -131,6 +135,14 @@
 
     static EmergencyParser::State emergency_state; // = EP_RESET
 
+    #if ENABLED(DIRECT_STEPPING)
```

Comment:
This all probably needs to be reordered. Probably should be 
1) RX Checks
2) R_UDR
3) Page manager
4) buffer_pos logic

Reply:
I'm not sure where to start with that. Those elements don't seem too neatly arranged here.

Reply:
I'll put in a separate commit for the re-order. mostly just want to get the page manager as high in the sequence as possible to shave off a few extra cycles per byte

Reply:
Pushed https://github.com/MarlinFirmware/Marlin/pull/17853/commits/f69ab9b5be6f2d3c65a282f0d5dc5035dc4bef70  

Should re-order that better (just does the buffer management stuff after the UART stuff)